### PR TITLE
Add Vue and React snippets to the dataviz widgets

### PR DESCRIPTION
**Pull request by AlexJameson**

### CODE REVIEW

File: api-reference/20 Data Visualization Widgets/BaseChart/1 Configuration/onPointClick.md [hunk 1]
```diff
@@ -57,7 +57,7 @@ The [onSeriesClick](/api-reference/20%20Data%20Visualization%20Widgets/dxChart/1
     import { Dx{WidgetName}Module } from "devextreme-angular";
     // ...
     export class AppComponent {
-        {widgetName}_onPointClick (e) {
+        onPointClick (e) {
```
Comment (RomanTsukanov) [inline api-reference/20 Data Visualization Widgets/BaseChart/1 Configuration/onPointClick.md]: Let's rename this function into `cancelSeriesClick` in examples for all the technologies.

File: api-reference/20 Data Visualization Widgets/dxChart/1 Configuration/argumentAxis/categories.md [hunk 2]
```diff
@@ -57,6 +57,55 @@ To sort the arguments, for example, alphabetically, you need to assign an array
         // ...
     })
 
+##### Vue
+
+    <!-- tab: App.vue -->
+    <template>
+        <DxChart ... >
+            <DxArgumentAxis 
+                :categories="['Africa', 'Antarctica', 'Asia', 'Australia',
```
Comment (RomanTsukanov) [inline api-reference/20 Data Visualization Widgets/dxChart/1 Configuration/argumentAxis/categories.md]: Let's declare this array as a component property in Angular, Vue, and React to make the samples more readable and avoid re-rendering in React.

File: api-reference/20 Data Visualization Widgets/dxChart/1 Configuration/onLegendClick.md [hunk 3]
```diff
@@ -12,7 +12,7 @@ A function that is executed when a [legend item](/concepts/05%20Widgets/Chart/35
 ##### param(e): Object
 Information about the event.
 
-##### field(e.component): {WidgetName}
+##### field(e.component): Chart
```
Comment (RomanTsukanov) [inline api-reference/20 Data Visualization Widgets/dxChart/1 Configuration/onLegendClick.md]: Unnecessary change.

File: api-reference/20 Data Visualization Widgets/dxChart/1 Configuration/onLegendClick.md [hunk 4]
```diff
@@ -57,7 +57,7 @@ The [onSeriesClick](/api-reference/20%20Data%20Visualization%20Widgets/dxChart/1
     import { DxChartModule } from "devextreme-angular";
     // ...
     export class AppComponent {
-        chart_onLegendClick (e) {
+        onLegendClick (e) {
```
Comment (RomanTsukanov) [inline api-reference/20 Data Visualization Widgets/dxChart/1 Configuration/onLegendClick.md]: `cancelSeriesClick` is a more suitable name in this case too.

File: api-reference/20 Data Visualization Widgets/dxPolarChart/1 Configuration/onLegendClick.md [hunk 5]
```diff
@@ -73,6 +73,53 @@ The [onSeriesClick](/api-reference/20%20Data%20Visualization%20Widgets/dxPolarCh
...
+
+    <script>
+    import DxPolarChart from 'devextreme-vue/polar-chart';
+
+    export default {
+        components: {
+            DxPolarChart
+        },
+        methods: {
+            onLegendClick (e) {
```
Comment (RomanTsukanov) [inline api-reference/20 Data Visualization Widgets/dxPolarChart/1 Configuration/onLegendClick.md]: And here.

File: api-reference/20 Data Visualization Widgets/dxSankey/1 Configuration/alignment.md [hunk 6]
```diff
@@ -43,6 +43,44 @@ A string value aligns all node columns uniformly. An array of strings allows you
...
+    <!-- tab: App.js -->
+    import React from 'react';
+
+    import Sankey from 'devextreme-react/sankey';
+
+    class App extends React.Component {
+        render() {
+            return (
+                <Sankey ... 
+                    alignment={['top', 'bottom', 'bottom']}>
```
Comment (RomanTsukanov) [inline api-reference/20 Data Visualization Widgets/dxSankey/1 Configuration/alignment.md]: In React, declare arrays as a component property or a constant/variable before the class declaration.

**Review by RomanTsukanov — CHANGES_REQUESTED** [review]

File: api-reference/20 Data Visualization Widgets/dxChart/1 Configuration/argumentAxis/categories.md [hunk 7]
```diff
@@ -48,6 +52,13 @@ To sort the arguments, for example, alphabetically, you need to assign an array
     // ...
     export class AppComponent {
         // ...
+        const countries = [
```
Comment (RomanTsukanov) [inline api-reference/20 Data Visualization Widgets/dxChart/1 Configuration/argumentAxis/categories.md]: `countries` is a component property. `const` is unnecessary.

File: api-reference/20 Data Visualization Widgets/dxChart/1 Configuration/argumentAxis/categories.md [hunk 8]
```diff
@@ -57,6 +68,72 @@ To sort the arguments, for example, alphabetically, you need to assign an array
...
+    } from 'devextreme-vue/chart'; 
+
+    export default {
+        components: {
+            DxChart,
+            DxArgumentAxis
+        },
+        data() {
+            return {
+                countries = [
```
Comment (RomanTsukanov) [inline api-reference/20 Data Visualization Widgets/dxChart/1 Configuration/argumentAxis/categories.md]:
```suggestion
                continents: [
```

File: api-reference/20 Data Visualization Widgets/dxChart/1 Configuration/argumentAxis/categories.md [hunk 9]
```diff
@@ -57,6 +68,72 @@ To sort the arguments, for example, alphabetically, you need to assign an array
...
+                    'Africa', 
+                    'Antarctica', 
+                    'Asia', 
+                    'Australia',
+                    'Europe'
+                ];
```
Comment (RomanTsukanov) [inline api-reference/20 Data Visualization Widgets/dxChart/1 Configuration/argumentAxis/categories.md]:
```suggestion
                ]
```

File: api-reference/20 Data Visualization Widgets/BaseChart/1 Configuration/onPointClick.md [hunk 10]
```diff
@@ -70,9 +70,56 @@ The [onSeriesClick](/api-reference/20%20Data%20Visualization%20Widgets/dxChart/1
...
+    <!-- tab: App.js -->
+    import React from 'react';
+
+    import {WidgetName} from 'devextreme-react/{widget-name}';
+
+    class App extends React.Component {
+        render() {
+            return (
+                <{WidgetName} ...
+                    onPointClick={cancelSeriesClick}>
```
Comment (RomanTsukanov) [inline api-reference/20 Data Visualization Widgets/BaseChart/1 Configuration/onPointClick.md]: `cancelSeriesClick` is defined in the context of the class, so you need to use `this` to address it.

File: api-reference/20 Data Visualization Widgets/dxChart/1 Configuration/argumentAxis/categories.md [hunk 11]
```diff
@@ -38,8 +43,7 @@ To sort the arguments, for example, alphabetically, you need to assign an array
 
     <!--HTML--><dx-chart ... >
         <dxo-argument-axis
-            [categories]="['Africa', 'Antarctica', 'Asia', 'Australia',
-                'Europe', 'North America', 'South America']">
+            [categories]="countries">
```
Comment (RomanTsukanov) [inline api-reference/20 Data Visualization Widgets/dxChart/1 Configuration/argumentAxis/categories.md]: These are continents, not countries.

File: api-reference/20 Data Visualization Widgets/dxChart/1 Configuration/argumentAxis/categories.md [hunk 12]
```diff
@@ -28,8 +28,13 @@ To sort the arguments, for example, alphabetically, you need to assign an array
         $("#chartContainer").dxChart({
             // ...
             argumentAxis: {
-                categories: ['Africa', 'Antarctica', 'Asia', 'Australia',
-                    'Europe', 'North America', 'South America']
+                categories:
```
Comment (RomanTsukanov) [inline api-reference/20 Data Visualization Widgets/dxChart/1 Configuration/argumentAxis/categories.md]: The opening square bracket is missing, so are the Americas.

File: api-reference/20 Data Visualization Widgets/dxChart/1 Configuration/argumentAxis/categories.md [hunk 13]
```diff
@@ -28,8 +28,13 @@ To sort the arguments, for example, alphabetically, you need to assign an array
...
             argumentAxis: {
-                categories: ['Africa', 'Antarctica', 'Asia', 'Australia',
-                    'Europe', 'North America', 'South America']
+                categories: 
+                    'Africa', 
+                    'Antarctica', 
+                    'Asia', 
+                    'Australia',
+                    'Europe'
+                ];
```
Comment (RomanTsukanov) [inline api-reference/20 Data Visualization Widgets/dxChart/1 Configuration/argumentAxis/categories.md]:
```suggestion
                ]
```

File: api-reference/20 Data Visualization Widgets/dxChart/5 Series Types/CommonSeries/aggregation/calculate.md [hunk 14]
```diff
@@ -76,4 +76,74 @@ One or several aggregated data objects. Should have the same structure as the or
...
+
+    export default {
+        components: {
+            DxChart,
+            DxSeries,
+            DxAggregation
+        },
+        methods: {
+            customAggregateFunc (aggregationInfo, series) {
+                let dataObjects = aggregationInfo.data;
```
Comment (RomanTsukanov) [inline api-reference/20 Data Visualization Widgets/dxChart/5 Series Types/CommonSeries/aggregation/calculate.md]:
```suggestion
                const dataObjects = aggregationInfo.data;
```

File: api-reference/20 Data Visualization Widgets/dxChart/5 Series Types/CommonSeries/aggregation/calculate.md [hunk 15]
```diff
@@ -76,4 +76,74 @@ One or several aggregated data objects. Should have the same structure as the or
...
+        }
+    }
+    </script>
+
+##### React
+
+    <!-- tab: App.js -->
+    import React from 'react';
+
+    import Chart from 'devextreme-react/chart';
```
Comment (RomanTsukanov) [inline api-reference/20 Data Visualization Widgets/dxChart/5 Series Types/CommonSeries/aggregation/calculate.md]: The `Series` and `Aggregation` imports are missing

File: api-reference/20 Data Visualization Widgets/dxChart/5 Series Types/CommonSeries/aggregation/calculate.md [hunk 16]
```diff
@@ -76,4 +76,74 @@ One or several aggregated data objects. Should have the same structure as the or
...
+
+    class App extends React.Component {
+        render() {
+            return (
+                <Chart ... >
+                    <Series ... >
+                        <DxAggregation>
```
Comment (RomanTsukanov) [inline api-reference/20 Data Visualization Widgets/dxChart/5 Series Types/CommonSeries/aggregation/calculate.md]:
```suggestion
                        <Aggregation
```

File: api-reference/20 Data Visualization Widgets/dxChart/5 Series Types/CommonSeries/aggregation/calculate.md [hunk 17]
```diff
@@ -76,4 +76,74 @@ One or several aggregated data objects. Should have the same structure as the or
...
+                            calculate={customAggregateFunc}
```
Comment (RomanTsukanov) [inline api-reference/20 Data Visualization Widgets/dxChart/5 Series Types/CommonSeries/aggregation/calculate.md]: `this` is required

File: api-reference/20 Data Visualization Widgets/dxChart/5 Series Types/CommonSeries/aggregation/calculate.md [hunk 18]
```diff
@@ -76,4 +76,74 @@ One or several aggregated data objects. Should have the same structure as the or
...
+                        />
+                    </Series ... >
```
Comment (RomanTsukanov) [inline api-reference/20 Data Visualization Widgets/dxChart/5 Series Types/CommonSeries/aggregation/calculate.md]:
```suggestion
                    </Series>
```

File: api-reference/20 Data Visualization Widgets/dxChart/5 Series Types/CommonSeries/aggregation/calculate.md [hunk 19]
```diff
@@ -76,4 +76,74 @@ One or several aggregated data objects. Should have the same structure as the or
...
+##### Vue
+
+    <!-- tab: App.vue -->
+    <template>
+        <DxChart ... >
+            <DxSeries ... >
+                <DxAggregation
+                    :calculate="customAggregateFunc"
+                />
+            </DxSeries ... >
```
Comment (RomanTsukanov) [inline api-reference/20 Data Visualization Widgets/dxChart/5 Series Types/CommonSeries/aggregation/calculate.md]:
```suggestion
            </DxSeries>
```

File: api-reference/20 Data Visualization Widgets/dxRangeSelector/1 Configuration/scale/aggregationInterval/aggregationInterval.md [hunk 20]
```diff
@@ -51,6 +51,60 @@ If the scale displays numbers, assign a number to this option. For example, an *
...
+        Scale,
+        AggregationInterval
+    } from 'devextreme-react/range-selector';
+
+    class App extends React.Component {
+        render() {
+            return (
+                <RangeSelector ... >
+                    <Scale
+                        aggregationInterval={"day"}>    {/* Interval of one day */}
```
Comment (RomanTsukanov) [inline api-reference/20 Data Visualization Widgets/dxRangeSelector/1 Configuration/scale/aggregationInterval/aggregationInterval.md]:
```suggestion
                        aggregationInterval="day">    {/* Interval of one day */}
```

**Review by RomanTsukanov — CHANGES_REQUESTED** [review]

File: api-reference/20 Data Visualization Widgets/dxChart/1 Configuration/argumentAxis/categories.md (hunk 12 above)
Comment (AlexJameson) [reply api-reference/20 Data Visualization Widgets/dxChart/1 Configuration/argumentAxis/categories.md]: Reverted the changes and added the missing square, thanks

File: api-reference/20 Data Visualization Widgets/dxChart/1 Configuration/argumentAxis/categories.md [hunk 21]
```diff
@@ -28,8 +28,15 @@ To sort the arguments, for example, alphabetically, you need to assign an array
...
-                categories: ['Africa', 'Antarctica', 'Asia', 'Australia',
-                    'Europe', 'North America', 'South America']
+                categories: [
+                    'Africa', 
+                    'Antarctica', 
+                    'Asia', 
+                    'Australia',
+                    'Europe',
+                    'North America',
+                    'South America',
```
Comment (RomanTsukanov) [inline api-reference/20 Data Visualization Widgets/dxChart/1 Configuration/argumentAxis/categories.md]:
```suggestion
                    'South America'
```

File: api-reference/20 Data Visualization Widgets/dxChart/1 Configuration/argumentAxis/categories.md [hunk 22]
```diff
@@ -48,6 +54,15 @@ To sort the arguments, for example, alphabetically, you need to assign an array
...
     export class AppComponent {
         // ...
+        continents = [
+            'Africa', 
+            'Antarctica', 
+            'Asia', 
+            'Australia',
+            'Europe',
+            'North America',
+            'South America',
```
Comment (RomanTsukanov) [inline api-reference/20 Data Visualization Widgets/dxChart/1 Configuration/argumentAxis/categories.md]:
```suggestion
            'South America'
```

File: api-reference/20 Data Visualization Widgets/dxChart/1 Configuration/argumentAxis/categories.md [hunk 23]
```diff
@@ -57,6 +72,76 @@ To sort the arguments, for example, alphabetically, you need to assign an array
...
+        data() {
+            return {
+                continents: [
+                    'Africa', 
+                    'Antarctica', 
+                    'Asia', 
+                    'Australia',
+                    'Europe',
+                    'North America',
+                    'South America',
```
Comment (RomanTsukanov) [inline api-reference/20 Data Visualization Widgets/dxChart/1 Configuration/argumentAxis/categories.md]:
```suggestion
                    'South America'
```

File: api-reference/20 Data Visualization Widgets/dxChart/1 Configuration/argumentAxis/categories.md [hunk 24]
```diff
@@ -57,6 +72,76 @@ To sort the arguments, for example, alphabetically, you need to assign an array
...
+    } from 'devextreme-react/chart';
+    
+    const continents = [
+        'Africa', 
+        'Antarctica', 
+        'Asia', 
+        'Australia',
+        'Europe',
+        'North America',
+        'South America',
```
Comment (RomanTsukanov) [inline api-reference/20 Data Visualization Widgets/dxChart/1 Configuration/argumentAxis/categories.md]:
```suggestion
        'South America'
```

File: api-reference/20 Data Visualization Widgets/dxChart/1 Configuration/argumentAxis/tickInterval/tickInterval.md [hunk 25]
```diff
@@ -46,6 +46,56 @@ If the axis displays numbers, assign a number to this option. If the axis displa
...
     })
 
+##### Vue
+
+    <!-- tab: App.vue -->
+    <template>
+        <DxChart ... >
+            <DxArgumentAxis>
+                <DxTickInterval :days="5" />
+            </DxArgumentAxis>>
```
Comment (RomanTsukanov) [inline api-reference/20 Data Visualization Widgets/dxChart/1 Configuration/argumentAxis/tickInterval/tickInterval.md]:
```suggestion
            </DxArgumentAxis>
```

File: api-reference/20 Data Visualization Widgets/dxChart/1 Configuration/valueAxis/minorTickInterval/minorTickInterval.md [hunk 26]
```diff
@@ -44,6 +44,56 @@ If the axis displays numbers, set the **minorTickInterval** to a number. This nu
...
     })
 
+##### Vue
+
+    <!-- tab: App.vue -->
+    <template>
+        <DxChart ... >
+            <DxValueAxis>
+                <DxMinorTickInterval :days="5" />
+            </DxValueAxis>>
```
Comment (RomanTsukanov) [inline api-reference/20 Data Visualization Widgets/dxChart/1 Configuration/valueAxis/minorTickInterval/minorTickInterval.md]:
```suggestion
            </DxValueAxis>
```

File: api-reference/20 Data Visualization Widgets/dxChart/1 Configuration/valueAxis/tickInterval/tickInterval.md [hunk 27]
```diff
@@ -44,6 +44,56 @@ If the axis displays numbers, set the **tickInterval** to a number. This number
...
     })
 
+##### Vue
+
+    <!-- tab: App.vue -->
+    <template>
+        <DxChart ... >
+            <DxValueAxis>
+                <DxTickInterval :days="5" />
+            </DxValueAxis>>
```
Comment (RomanTsukanov) [inline api-reference/20 Data Visualization Widgets/dxChart/1 Configuration/valueAxis/tickInterval/tickInterval.md]:
```suggestion
            </DxValueAxis>
```

File: api-reference/20 Data Visualization Widgets/dxChart/5 Series Types/CommonSeries/aggregation/calculate.md [hunk 28]
```diff
@@ -76,4 +76,77 @@ One or several aggregated data objects. Should have the same structure as the or
...
+                            calculate={this.customAggregateFunc}
+                        />
+                    </Series>
+                </Chart>
+            );
+        }
+
+        customAggregateFunc (aggregationInfo, series) {
+            let dataObjects = aggregationInfo.data;
+            let result = { }; {/* or [ ] */}
```
Comment (RomanTsukanov) [inline api-reference/20 Data Visualization Widgets/dxChart/5 Series Types/CommonSeries/aggregation/calculate.md]:
```suggestion
            let result = { }; // or [ ]
```

File: api-reference/20 Data Visualization Widgets/dxChart/5 Series Types/CommonSeries/aggregation/calculate.md [hunk 29]
```diff
@@ -76,4 +76,77 @@ One or several aggregated data objects. Should have the same structure as the or
...
+
+            {/* Aggregate the data objects here */}
```
Comment (RomanTsukanov) [inline api-reference/20 Data Visualization Widgets/dxChart/5 Series Types/CommonSeries/aggregation/calculate.md]:
```suggestion
            // Aggregate the data objects here
```

File: api-reference/20 Data Visualization Widgets/dxFunnel/1 Configuration/onItemClick.md [hunk 30]
```diff
@@ -72,6 +72,53 @@ This function is often used to implement item selection as shown in the followin
...
+    <!-- tab: App.js -->
+    import React from 'react';
+
+    import Funnel from 'devextreme-react/funnel';
+
+    class App extends React.Component {
+        render() {
+            return (
+                <Funnel ...
+                    onItemClick={selectItem}>
```
Comment (RomanTsukanov) [inline api-reference/20 Data Visualization Widgets/dxFunnel/1 Configuration/onItemClick.md]:
```suggestion
                    onItemClick={this.selectItem}>
```

File: api-reference/20 Data Visualization Widgets/dxPolarChart/1 Configuration/onLegendClick.md [hunk 31]
```diff
@@ -70,9 +70,56 @@ The [onSeriesClick](/api-reference/20%20Data%20Visualization%20Widgets/dxPolarCh
     })
 
     <!--HTML--><dx-polar-chart ...
-        (onLegendClick)="polarChart_onLegendClick($event)">
+        (onLegendClick)="cancelSeriesClick ($event)">
```
Comment (RomanTsukanov) [inline api-reference/20 Data Visualization Widgets/dxPolarChart/1 Configuration/onLegendClick.md]:
```suggestion
        (onLegendClick)="cancelSeriesClick($event)">
```

File: api-reference/20 Data Visualization Widgets/dxRangeSelector/1 Configuration/scale/aggregationInterval/aggregationInterval.md [hunk 32]
```diff
@@ -51,6 +51,60 @@ If the scale displays numbers, assign a number to this option. For example, an *
...
+##### Vue
+
+    <!-- tab: App.vue -->
+    <template>
+        <DxRangeSelector ... >
+            <DxScale
+                aggregation-interval="day">     <!-- Interval of one day -->
+                <DxAggregationInterval 
+                    :days="5" />                <!-- Interval of five days -->
+            </DxScale>>
```
Comment (RomanTsukanov) [inline api-reference/20 Data Visualization Widgets/dxRangeSelector/1 Configuration/scale/aggregationInterval/aggregationInterval.md]:
```suggestion
            </DxScale>
```

File: api-reference/20 Data Visualization Widgets/dxTreeMap/1 Configuration/onClick.md [hunk 33]
```diff
@@ -72,6 +72,53 @@ This function is often used to implement item selection as shown in the followin
...
+    <!-- tab: App.js -->
+    import React from 'react';
+
+    import TreeMap from 'devextreme-react/tree-map';
+
+    class App extends React.Component {
+        render() {
+            return (
+                <TreeMap ...
+                    onClick={selectItem}>
```
Comment (RomanTsukanov) [inline api-reference/20 Data Visualization Widgets/dxTreeMap/1 Configuration/onClick.md]:
```suggestion
                    onClick={this.selectItem}>
```

File: api-reference/20 Data Visualization Widgets/dxChart/1 Configuration/argumentAxis/aggregationInterval/aggregationInterval.md [hunk 34]
```diff
@@ -51,6 +51,60 @@ If the axis displays numbers, assign a number to this option. For example, an **
...
+##### Vue
+
+    <!-- tab: App.vue -->
+    <template>
+        <DxChart ... >
+            <DxArgumentAxis
+                aggregation-interval="day">     <!-- Interval of one day -->
+                <DxAggregationInterval 
+                    :days="5" />                <!-- Interval of five days -->
+            </DxArgumentAxis>>
```
Comment (RomanTsukanov) [inline api-reference/20 Data Visualization Widgets/dxChart/1 Configuration/argumentAxis/aggregationInterval/aggregationInterval.md]:
```suggestion
            </DxArgumentAxis>
```

File: api-reference/20 Data Visualization Widgets/dxChart/1 Configuration/argumentAxis/minorTickInterval/minorTickInterval.md [hunk 35]
```diff
@@ -46,6 +46,56 @@ If the axis displays numbers, assign a number to this option. If the axis displa
...
     })
 
+##### Vue
+
+    <!-- tab: App.vue -->
+    <template>
+        <DxChart ... >
+            <DxArgumentAxis>
+                <DxMinorTickInterval :days="5" />
+            </DxArgumentAxis>>
```
Comment (RomanTsukanov) [inline api-reference/20 Data Visualization Widgets/dxChart/1 Configuration/argumentAxis/minorTickInterval/minorTickInterval.md]:
```suggestion
            </DxArgumentAxis>
```

**Review by RomanTsukanov — APPROVED** [review]
